internal/gateway/gateway_test.go

```diff
@@ -95,6 +95,56 @@ func TestGateway(t *testing.T) {
 	gw2 := gateways[1]
 	gw2.ServeHTTP(rec, req)
 	assert.Equal(t, http.StatusUnauthorized, rec.Result().StatusCode)
+
+	// make request for allowUnauthorized endpoint without a token
```

Comment:
Can you add another test for `make request for endpoint with `allowUnauthorized=false` without a token` which returns unauthorized